### ISSUE TITLE
Improve Search form: correct some invalid markup and various bits of missing or ungrammatical text

### ISSUE THREAD

**Issue by AlOneill** [FreeUKGen/FreeBMD2#314]
I hesitate to make any negative comments about such a magnificent implementation of what is a very complicated Search form — excellent work! However, there are a few little problems that should be addressed:

*  there are 2 instances of `div` elements being child elements of an `ol`. This is invalid markup and will cause problems for screen-reader users: `div.recordTypeGrid` should be a `li` as should the `div` that is the parent of `label.search_query_bmd_record_type_marriage`. The closing tags are already in place.

*  a space is missing in the Surname tooltip text between the full-stop and 'Check': "… lower-case.Check … "

*  the Record Type tooltip text does not read well (ungrammatical). Also, why is the info icon black, unlike the others?

*  the Age at death tooltip text appears to have some words missing

*  the tooltip for Counties is present but empty. There is, however, info under the select box

*  the 'Spouses surname' info icon is present but there is nothing there, not even an empty tooltip. Also, to match the label for 'Spouse first name(s)', I suggest we use 'Spouse surname' as the label rather than 'Spouses surname' (which would need an apostrophe)

We _could_ be criticised for not using `fieldset` elements, but they would not be simple to implement, even 'invisible' ones (for screen-reader users only).

**Comment by Vino-S** [comment]
Ready for testing in test2

**Comment by richpomfret** [comment]
@AlOneill to test.

**Comment by AlOneill** [comment]
@Vino-S The list issues have been fixed. Thank you.

The validator is complaining that: "The value of the `for` attribute of the `label` element must be the ID of a non-hidden form control." There are four instances:

The first instance is where the `label` text is "Date range from" and where `for="start_quarter"` but the form control — the `select` box — appears to have `id="from_quarter"`.

The second is where the `label` text is "Date range to" and where `for="to_year"`.

The third is where the `label` is "Record type", and the fourth where the `label` is "Counties".

These seem like valid complaints as screen-reader software uses `for` and `id` to connect label text with form controls, but I could be missing something!

(The labels and tooltips have been changed, but not as I expected and with some inconsistencies — will get back to you on those.)

**Comment by richpomfret** [comment]
@Vino-S to check before closing.

**Comment by Vino-S** [comment]
Ready for testing in test3

**Comment by AlOneill** [comment]
@Vino-S  Apologies for the delay here.

The text below the **County select** says 'Select one County or none' whereas at least two counties can be selected. Is there a limit on how many counties can be selected?

Suggest we edit the text to be similar to that for the District select, as appropriate.

The tooltip text for each pair of **date selects** does not make sense. Currently each is: Leave Year blank to search the whole year. Suggest we do not need a tool-tip on these.

The **Age at death** tooltip is not good: "Age can also be entered Number(range) or Date of Birth(range)".
The options are: Age, Age range, Year of birth, Year of birth range and these seem to be self-explanatory, so I suggest that the tooltip is not needed.

The tooltips for **First Names and Surname** apply to the next input (the check-box) for each, and not the type-here inputs which belong to those labels. I think this would be confusing for a screen-reader user. I suggest we delete the tooltips.

@PatReynolds @DeniseColbert What do you think about deleting these tooltips?

**Comment by AlOneill** [comment]
@Vino-S I'm having a hard time interpreting the results of the validator testing: too much noise.

But, from tabbing through the form and inspecting the markup there are now two types of problem:

1. empty tooltips for Record type, Volume, Page, Counties, Districts

2. labels that have `tabindex="0"` but no associated tooltip — so there is nothing to see on tabbing: Spouse First Name(s), Spouse surname, Choose the Name field, Select one of the following

I think (hope) that is all.

**Comment by AlOneill** [comment]
@Vino-S 
Arguably, the info below the County and District select boxes should be heard by a screen-reader user _before_ they make their selections. In which case, a tooltip would seem the obvious choice.

If we decide that we need to remind users how to select more than one County or District, then the tooltip would need revising to accommodate this. Sigh.
 (Apologies for not thinking this through before I commented yesterday.)

**Comment by richpomfret** [comment]
@Vino-S to update based on above findings.

**Comment by Vino-S** [comment]
Following are the changes:
1. Moved the text for County and District to just below the label. Added help text for multi-select
2. Removed tooltips for First Name, Surname, Volume, Page, County, Record Type, Age at Death, Districts, Spouse First Name, and Spouse Surname

**Comment by richpomfret** [comment]
@AlOneill to take a look.

**Comment by AlOneill** [comment]
There is some leftover markup related to tooltips that are not in use (blind tabbing is the result): `search_query_chapman_codes` and `search_query_districts`, also `spouses_mother_surname`, Spouse first name(s) and Spouse surname.

The Spouse first name(s) and Spouse surname inputs have lost their you-are-here styling.

The text below the Districts label says: Select one/more Districts(hold Ctrl for multiple) or none
Screen-reader uses might not be aware of punctuation characters, and these can mean different things depending on the context; and there is a missing space. I suggest: Select none, one or more Districts (hold Ctrl for multiple)

Let's assume that Mac users know what to do, unless we hear otherwise (!).

Cannot tab into Age at Death field — it has a `disabled` attribute even where 'Death' record type is the sole choice.

**Comment by Vino-S** [comment]
ready for testing in test2

**Comment by richpomfret** [comment]
@AlOneill can you take a look?

**Comment by AlOneill** [comment]
@Vino-S There are still a couple of empty tabs: between Page input and Counties input and again between Counties input and Districts input.

If I select say Births and then select Deaths as the record type, I'm unable to tab into "Age at death" input, neither can I click into it. It works OK if I select Deaths initially, but otherwise the `disabled=""` gets left behind!

**Comment by AlOneill** [comment]
@Vino-S The label "Mothers surname" should be "Mother's surname" or possibly "Mother surname" which would match "Spouse first name(s)".

Similarly "Spouses surname" should be either "Spouse's surname" or "Spouse surname".

The style of the label text should be consistent across all fields and as the placeholders use apostrophes, I suggest the other labels be adjusted to match the placeholder text. Thank you.

**Comment by Vino-S** [comment]
Ready for testing in test3

**Comment by PatReynolds** [comment]
@PatReynolds  @DeniseColbert to test

**Comment by AlOneill** [comment]
@Vino-S The apostrophes have been sorted — thank you!

There is a stray `ttip` class on the "Spouse's Surname" label, together with a `tabindex="0"` — this produces an empty tab.

The other empty tabs have been removed — thank you.

**Comment by Vino-S** [comment]
@AlOneill 
Removed the tabindex on Spouse Surname label. Ready for testing on test2

**Comment by AlOneill** [comment]
@Vino-S Spouse surname label now behaves as expected (but I note that the ttip class remains).
